include/spirv-tools/libspirv.h

```diff
@@ -346,7 +346,7 @@ typedef spv_context_t* spv_context;
 // Certain target environments impose additional restrictions on SPIR-V, so it's
 // often necessary to specify which one applies.  SPV_ENV_UNIVERSAL means
 // environment-agnostic SPIR-V.
-enum spv_target_env { SPV_ENV_UNIVERSAL, SPV_ENV_VULKAN };
+enum spv_target_env { SPV_ENV_UNIVERSAL_1_0_4, SPV_ENV_VULKAN_1_0_7 };
```

Comment:
It's great having this level of control.  But right now Vulkan is rev'ing at a weekly pace.
Many users will only care about a more coarse grain target.  That is I expect uses that drop the revision number, e.g.
SPV_ENV_UNIVERSAL_1_0 and SPV_ENV_VULKAN_1_0 will be very common.

So lets add those *in addition* to the more specific ones.  The people who care about the tiniest changes can still get what they need.

Also the particular enum values should be stable over time.  So don't make the less specific ones equal to the more specific ones.

I think that also means that even when we retire support for a very specific target env enum, we have to keep its value slot occupied.   But we could change its name to break clients to indicate that the code is no longer supported. We could also cause the context creation to return nullptr when we don't support that value any more.